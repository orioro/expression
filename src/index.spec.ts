import * as api from './index'

test('public api', () => {
  expect(Object.keys(api)).toMatchSnapshot()
})
