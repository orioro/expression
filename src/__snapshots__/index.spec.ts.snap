// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`public api 1`] = `
Array [
  "ALL_EXPRESSIONS",
  "$$INDEX",
  "$$ARRAY",
  "$$ACC",
  "$$SORT_A",
  "$$SORT_B",
  "$arrayIncludes",
  "$arrayIncludesAll",
  "$arrayIncludesAny",
  "$arrayLength",
  "$arrayReduce",
  "$arrayMap",
  "$arrayEvery",
  "$arraySome",
  "$arrayFilter",
  "$arrayFindIndex",
  "$arrayIndexOf",
  "$arrayFind",
  "$arrayReverse",
  "$arraySort",
  "$arrayPush",
  "$arrayPop",
  "$arrayUnshift",
  "$arrayShift",
  "$arraySlice",
  "$arrayReplace",
  "$arrayAddAt",
  "$arrayRemoveAt",
  "$arrayJoin",
  "$arrayAt",
  "ARRAY_EXPRESSIONS",
  "$boolean",
  "BOOLEAN_EXPRESSIONS",
  "$eq",
  "$notEq",
  "$in",
  "$notIn",
  "$gt",
  "$gte",
  "$lt",
  "$lte",
  "$matches",
  "COMPARISON_EXPRESSIONS",
  "$pipe",
  "FUNCTIONAL_EXPRESSIONS",
  "$and",
  "$or",
  "$not",
  "$nor",
  "$xor",
  "$if",
  "$switch",
  "$switchKey",
  "LOGICAL_EXPRESSIONS",
  "$mathSum",
  "$mathSub",
  "$mathMult",
  "$mathDiv",
  "$mathMod",
  "$mathPow",
  "$mathAbs",
  "$mathMax",
  "$mathMin",
  "$mathRound",
  "$mathFloor",
  "$mathCeil",
  "MATH_EXPRESSIONS",
  "$numberInt",
  "$numberFloat",
  "NUMBER_EXPRESSIONS",
  "$objectMatches",
  "$objectFormat",
  "$objectDefaults",
  "$objectAssign",
  "$objectKeys",
  "OBJECT_EXPRESSIONS",
  "$string",
  "$stringStartsWith",
  "$stringLength",
  "$stringSubstr",
  "$stringConcat",
  "$stringTrim",
  "$stringPadStart",
  "$stringPadEnd",
  "$stringToUpperCase",
  "$stringToLowerCase",
  "$stringInterpolate",
  "STRING_EXPRESSIONS",
  "CORE_TYPES",
  "typeExpressions",
  "$type",
  "$isType",
  "TYPE_EXPRESSIONS",
  "$$VALUE",
  "$value",
  "$literal",
  "$evaluate",
  "VALUE_EXPRESSIONS",
  "TypeAlternatives",
  "TypeMap",
  "isExpression",
  "evaluate",
  "evaluateTyped",
  "evaluateTypedAsync",
  "syncInterpreter",
  "syncInterpreterList",
  "asyncInterpreter",
  "asyncInterpreterList",
]
`;
